feature_engine/disc_features.ipynb

```diff
@@ -0,0 +1,2134 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 45,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import numpy as np"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 46,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def preprocess_conversation_columns(df):\n",
+    "\t# remove all special characters from df\n",
+    "\tdf.columns = df.columns.str.replace('[^A-Za-z0-9_]', '', regex=True)\n",
+    "\t\n",
+    "\t# If data is grouped by batch/round, add a conversation num\n",
+    "\tif {'batch_num', 'round_num'}.issubset(df.columns):\n",
+    "\t\tdf['conversation_num'] = df.groupby(['batch_num', 'round_num']).ngroup()\n",
+    "\t\tdf = df[df.columns.tolist()[-1:] + df.columns.tolist()[0:-1]] # make the new column first\n",
+    "\n",
+    "\treturn(df)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 47,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "test_dd = preprocess_conversation_columns(pd.read_csv('./data/raw_data/test_dd.csv'))\n",
+    "test_dd_vect = pd.read_csv('./embeddings/test_dd.csv')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 48,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "csop = preprocess_conversation_columns(pd.read_csv('./data/raw_data/csopII_conversations_withblanks.csv'))\n",
+    "vect = pd.read_csv('./embeddings/csopII_conversations_withblanks.csv')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 49,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from features.get_all_DD_features import *"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 50,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "test_dd['message_embedding'] = conv_to_float_arr(test_dd_vect['message_embedding'].to_frame())\n",
+    "csop['message_embedding'] = conv_to_float_arr(vect['message_embedding'].to_frame())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 56,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# look at a unit case just for testing\n",
+    "csop = csop[csop[\"conversation_num\"].isin([\"25PJSrbtr8mkHTvcw_easy\",\"cb2KMXFqSt2fJqRoa_hard\"])]"
```

Comment:
Here, I'm testing two specific conversations:

- `25PJSrbtr8mkHTvcw_easy` has only 4 chats in it (so it has only 2 "chunks" even when we ask it to give us 3);
- `cb2KMXFqSt2fJqRoa_hard` has more chats, so it gives us the full 3 chunks.